pkgs/by-name/zi/zitadel/package.nix

```diff
@@ -72,20 +101,31 @@ let
       nativeBuildInputs = nativeBuildInputs ++ [
         buf
         cacert
+        yq-go
       ];
 
       buildPhase = ''
+        runHook preBuild
+
+        yq --inplace '.deps = []' proto/buf.yaml
+        yq --inplace '.deps = []' proto/buf.lock
+        yq eval-all --inplace '[ select(fileIndex == 0),select(fileIndex == 1) ] | .[0].directories += .[1] | .[0]' buf.work.yaml - <<"END"
+        ${builtins.toJSON (builtins.attrNames generatedSrc.protobufDeps)}
+        END
+        ${lib.concatLines (
+          lib.flip lib.mapAttrsToList generatedSrc.protobufDeps (
+            name: dep: "cp -r ${fetchProtobufDep dep} ${name}"
+          )
+        )}
```

Comment:
This is clever piece of code, but could we make all protobuf deps into a single `symlinkJoin` (or `linkFarm` if you want to rename them)? And then simply copy everything from that into the builder. Also, symlinks are preferable, but I assume zitadel doesn't like that?

Reply:
Issue is that this needs to be inside the source tree, which means (unless I'm mistaken about how it works) *symlinkJoin* and *linkFarm* won't help much.
The former because it's built to merge output trees and it would need to happen as part of passing *src*, and therefore would make the `yq --inplace` work on symlinks and I'm not sure how well that'd work (it might try to write to the target which'd err out with EPERM).
The latter would just create a tree which'd still need to be merged into *src* which again sounds tricky.

I'll see if it still builds with symlinks since that's a cheap `ln -s` rather than `cp -r` which might speed up builds.

(if I'm mistaken about anything above *please* tell me ^^)

Reply:
The `ln -s` version builds nicely, so I'll resolve this, feel free to reopen though if you can think of a different approach that I failed to see (especially in regards to *symlinkJoin* and *linkFarm*.

Reply:
My idea was to do something like this
```nix
let
  ...
  protobufDeps = linkFarm "zitadel-protobuf-deps" (lib.mapAttrsToList (name: dep: { inherit name; path = fetchProtobufDep dep; }) generatedSrc.protobufDeps);
  ...
  cp -r "${protobufDeps}"/* ./
```
So we don't have to end up generating bash. But it's not the end of the world and if you don't like that I'd at least more like the inner expression to be extracted into a variable so the code to parse is less dense